[FSDP] restricts meta model check to non ignored modules in FSDP

colin2328 opened the pull request:
Summary: as title

Test Plan: see test plan D40287799

Differential Revision: D40287890

awgu reviewed (approved):
LGTM!

rohan-varma reviewed (commented):
We should ideally include an OSS unit test that would've caught this issue, otherwise LGTM. Thanks for the fix!

rohan-varma commented on torch/distributed/fsdp/fully_sharded_data_parallel.py:
```diff
@@ -1367,7 +1368,7 @@ def _materialize_module(
         ``deferred_init.materialize_module()``.
         """
-        is_meta_module = any(p.is_meta for p in module.parameters())
+        is_meta_module = any(p.is_meta for p in self._get_orig_params(module, ignored_params))
```
nit: the fake.is_fake check on 1375 may also have to be updated. But feel free to file an issue if not in scope of this PR.

colin2328 committed:
[FSDP] restricts meta model check to non ignored modules in FSDP (#86766)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/86766

as title

Test Plan: see test plan D40287799

Differential Revision: D40287890

fbshipit-source-id: ffc4c57102a0dd0ccd6e536d050b073fd4baa565

colin2328 commented:
/easycla

colin2328 commented:
@pytorchbot merge -g

colin2328 commented:
/easycla

colin2328 commented:
@pytorchbot merge -g

colin2328 commented:
/easycla

colin2328 commented:
@pytorchbot merge -g

rohan-varma commented:
@colin2328 @awgu Could we get a UT added here + follow up on the fake tensor thing as well

colin2328 commented:
@pytorchbot label "topic: bug fixes"
